File: docs/sistemas/PagamentoUnico.js
import AmortizationSystem from "./AmortizationSystem.js";

export default class PagamentoUnico extends AmortizationSystem {
    constructor() {
        super("Sistema de Pagamento Único");
    }
    atualizarValores(montante, tempo, taxa) {
        this.juros = this.saldo * taxa;
        this.saldo += this.juros;
    }
    ultimaParcela(montante, tempo, taxa) {
        this.atualizarValores(montante, tempo, taxa);
        this.amortizacao = montante;
        this.pagamento = this.saldo;
    }
}

File: src/main/java/model/PagamentoUnico.java
package model;

public class PagamentoUnico extends Amortizacao {

    @Override
    protected void atualizarValores(Emprestimo loan) {
        juros = saldo * loan.getTaxa();
        saldo += juros;
    }

    @Override
    protected void ultimaParcela(Emprestimo loan) {
        atualizarValores(loan);
        amortizacao = loan.getMontante();
        pagamento = saldo;
    }
}
